[rb] initialize all of the drivers in superclass

titusfortner opened the pull request:
All of the drivers are now pretty much doing the exact same thing, so there isn't a need to have the code duplicated in each file.

The only major issue I had was extending the browser-specific Bridge modules. This code changes them into subclasses, which works, but I don't love, especially because it is adding files that don't actually do anything (right now), just so that the superclass can  make all the method calls it needs. 

Should we define the subclasses at the end of `remote/bridge.rb` file rather than creating 2 extra files? Or is the a better approach entirely?

p0deje commented on rb/lib/selenium/webdriver/common/driver.rb:
```diff
@@ -71,9 +71,28 @@ def for(browser, opts = {})
+          default_caps = @bridge&.browser || :new
+          desired_capabilities = opts.delete(:desired_capabilities) || Remote::Capabilities.send(default_caps)
+          if desired_capabilities.is_a?(Symbol)
+            unless Capabilities.respond_to?(desired_capabilities)
```
This should be `Remote::Capabilities`.

p0deje commented on rb/lib/selenium/webdriver/common/driver.rb:
```diff
@@ -71,9 +71,28 @@ def for(browser, opts = {})
+              raise Error::WebDriverError, "invalid desired capability: #{desired_capabilities.inspect}"
+            end
+
+            desired_capabilities = Capabilities.__send__(desired_capabilities)
```
Same: `Remote::Capabilities`.

p0deje commented on rb/lib/selenium/webdriver/edge_html/bridge.rb:
```diff
@@ -0,0 +1,29 @@
+    module EdgeHtml
+      class Bridge < WebDriver::Remote::Bridge
+      end # Bridge
+    end # EdgeChrome
```
EdgeHtml

titusfortner commented:
@p0deje I implemented it with the approach you suggested, this is what I came away with:

```ruby
namespacing = self.class.to_s.split('::')

if defined?(Object.const_get("#{namespacing[0, namespacing.length - 1].join('::')}::Bridge")) &&
          !namespacing.include?('Remote')
  bridge.extend Object.const_get("#{namespacing[0, namespacing.length - 1].join('::')}::Bridge")
end
```

I think I like it better, even though this code is ugly.

twalpole commented:
@titusfortner You shouldn't need the "namespace.length - 1" stuff - you should be able to use range with a negative index `namespacing[0..-2]`

twalpole commented:
Also I think you probably want to use `const_defined?`

    if Object.const_defined?(namespacing[0..-2].push("Bridge").join('::')) && !namespacing.include?('Remote')

titusfortner commented:
Thanks @twalpole, those changes make it cleaner.
